zio-http/src/main/scala/zhttp/http/auth/Partials.scala

```diff
@@ -0,0 +1,55 @@
+package zhttp.http.auth
+
+import zhttp.http.{ Http, HttpError, Request, Response, ResponseM }
+import zio.ZIO
+
+object Partials {
+  private[auth] final case class CollectAuthedPartiallyApplied[A](private val dummy: Boolean = false)
+    extends AnyVal {
+
+    def apply[R, E](
+      pf: PartialFunction[AuthedRequest[A], Response[R, E]]
+    ): Http[R, E, AuthedRequest[A], Response[R, E]] =
+      Http.identity[AuthedRequest[A]] >>= { a =>
+        if (pf.isDefinedAt(a))
+          Http.succeed(pf(a))
+        else
+          Http.empty
+      }
+  }
+
+  private[auth] final case class CollectAuthedMPartiallyApplied[A](private val dummy: Boolean = false)
+    extends AnyVal {
+
+    def apply[R, E](
+      pf: PartialFunction[AuthedRequest[A], ResponseM[R, E]]
+    ): Http[R, E, AuthedRequest[A], Response[R, E]] =
+      Http.identity[AuthedRequest[A]] >>= { a =>
+        if (pf.isDefinedAt(a))
+          Http.fromEffect(pf(a))
+        else
+          Http.empty
+      }
+  }
+
+  private[auth] final case class HttpAuthenticationMiddlewarePartiallyApplied[A](private val dummy: Boolean = false)
+    extends AnyVal {
+
+    def apply[R, E <: HttpError, A1 >: A](
+      f: Request => Either[E, AuthedRequest[A1]]
```

Comment:
```suggestion
      f: Request => Http[R, E, AuthedRequest[A1]]
```

Reply:
It's missing the `A` type parameter in `Http` if I do that, do  you mean replacing `Request => Either[E, AuthedRequest[A1]]` by `Http[E, Request, AuthedRequest[A1]]`?

Reply:
Fixed the comment.